llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -2518,6 +2519,19 @@ void SITargetLowering::allocatePreloadKernArgSGPRs(
       unsigned NumAllocSGPRs =
           alignTo(ArgLoc.getLocVT().getFixedSizeInBits(), 32) / 32;
 
+      if (!AlignedForImplictArgs && Arg.getName().starts_with("_hidden")) {
```

Comment:
Should be expecting inreg attribute

Reply:
This is lowering both explicit and implicit arguments now, but only the ones marked with inreg. The check here is if extra padding SGPRs are needed when the implicitargs start. The firmware copies the data to SGPRs directly from the kernarg segment. With the alignment of the kernarg segment we may need to allocate an empty SGPR. We would not like to allocate it this way but need to understand that the HW is and we don't really have a choice which SGPR gets which data.